AK/Optional.h

```diff
@@ -48,8 +49,135 @@ struct OptionalNone {
     explicit OptionalNone() = default;
 };
 
+template<typename T, typename Self>
```

Comment:
```suggestion
template<typename T, typename Self = Optional<T>>
```
Should make the inheritance a bit nicer

Reply:
Also moving this commit up would be nice, as that removes a lot of "added then immediately removed code"

Reply:
I did it like this in case `OptionalBase` had to be reverted, but as I didn't see any objections to it, and since seems to work fine now, I can do this.